src/main/java/com/exactpro/th2/fix/client/util/FixBeanUtil.java

```diff
@@ -1,36 +1,33 @@
 package com.exactpro.th2.fix.client.util;
 
+import com.exactpro.th2.fix.client.Main;
 import com.exactpro.th2.fix.client.fixBean.FixBean;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.StandardOpenOption;
-import java.util.List;
 
 public class FixBeanUtil {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FixBeanUtil.class);
+    public static File createConfig(Main.Settings settings) throws IOException {
 
-    public static File createConfig(List<FixBean> fixBeans) {
         StringBuilder sb = new StringBuilder();
 
-        sb.append(fixBeans.get(0).toConfig("default"));
+        sb.append(settings.toConfig("default"));
 
-        for (FixBean fixBean : fixBeans) {
+        for (FixBean fixBean : settings.getSessionsSettings()) {
             sb.append(fixBean.toConfig("session"));
         }
-        File configFile = null;
+
+        File configFile;
         try {
             configFile = File.createTempFile("config", ".cfg");
-            Files.writeString(Path.of(configFile.getAbsolutePath()), sb.toString(), StandardOpenOption.APPEND);
+            Files.writeString(Path.of(configFile.getAbsolutePath()), sb.toString());
         } catch (IOException e) {
-            LOGGER.error("Failed to create config file", e);
-            System.exit(1);
+            throw new IOException("Failed to create config file.", e);
```

Comment:
Add file name to exception message. Also maybe replace `IOException` with `IllegalStateException`